mlir-tensorrt/compiler/lib/Compiler/OptionsProviders.cpp

```diff
@@ -0,0 +1,64 @@
+//===- OptionsProviders.cpp -------------------------------------*- C++ -*-===//
+//
+// SPDX-FileCopyrightText: Copyright 2024 NVIDIA CORPORATION & AFFILIATES.
+// All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+//===----------------------------------------------------------------------===//
+///
+/// Data structures and functions for manipulating compiler options.
+///
+//===----------------------------------------------------------------------===//
+#include "mlir-tensorrt/Compiler/OptionsProviders.h"
+#include "cuda_runtime_api.h"
+#include "llvm/Support/Error.h"
+
+// TODO (pranavm): Check if we can just reuse `DeviceInfo.cpp`?
```

Comment:
This logic looks almost the same as what's in `DeviceInfo.cpp`. If we can share code between them that might be good.

Reply:
@christopherbate I guess the question here is whether the compiler is allowed to depend on the executor?

Reply:
Yes, compiler depends on the `executor` project

Reply:
Declaring dependency on `MLIRTensorRTSupportDeviceInfo` is fine